src/db/options.rs

```diff
@@ -1,28 +1,59 @@
 //! Options for database-level commands.
-use bson::Document;
+use bson::{Bson, Document};
 use common::WriteConcern;
 use db::roles::Role;
 
 #[derive(Default)]
 pub struct CreateCollectionOptions {
-    pub capped: bool,
-    pub auto_index_id: bool,
+    pub capped: Option<bool>,
+    pub auto_index_id: Option<bool>,
     pub size: Option<i64>,
     pub max: Option<i64>,
-    pub use_power_of_two_sizes: bool,
-    pub no_padding: bool,
+    pub use_power_of_two_sizes: Option<bool>,
+    pub no_padding: Option<bool>,
 }
 
 impl CreateCollectionOptions {
     pub fn new() -> CreateCollectionOptions {
-        CreateCollectionOptions {
-            capped: false,
-            auto_index_id: true,
-            size: None,
-            max: None,
-            use_power_of_two_sizes: true,
-            no_padding: false,
+        Default::default()
+    }
+}
+
+impl From<CreateCollectionOptions> for Document {
+    fn from(options: CreateCollectionOptions) -> Self {
+        let mut document = Document::new();
+
+        if let Some(capped) = options.capped {
+            document.insert("capped", Bson::Boolean(capped));
+        }
+
+        if let Some(auto_index_id) = options.auto_index_id {
+            document.insert("autoIndexId", Bson::Boolean(auto_index_id));
+        }
+
+        if let Some(size) = options.size {
+            document.insert("size", Bson::I64(size));
+        }
+
+        if let Some(max) = options.max {
+            document.insert("max", Bson::I64(max));
+        }
+
+        let mut flags = 0;
+
+        if let Some(true) = options.use_power_of_two_sizes {
```

Comment:
Perhaps another place where we could use `bitflags`? Not sure if it's worth it for something so small or not